introspection/parse.go

```diff
@@ -26,10 +26,18 @@ func ParseIntrospectionQuery(query Query) *ast.SchemaDocument {
 func parseSchemaDefinition(query Query, typeMap map[string]*FullType) *ast.SchemaDefinition {
 	def := ast.SchemaDefinition{}
 
-	def.OperationTypes = append(def.OperationTypes,
-		parseOperationTypeDefinitionForQuery(typeMap[*query.Schema.QueryType.Name]),
-		parseOperationTypeDefinitionForMutation(typeMap[*query.Schema.MutationType.Name]),
-	)
+	// Is it possible to have an empty QueryType? I assume it would be possible but I'm not sure
```

Comment:
No is not. Query must have in schema.

Reply:
Done

Reply:
And yeah I succeed fixing it. The big bunch of `introspection/type.go:75:6: type `DirectiveType` is unused (unused)` ones appears only locally, not sure why.

To be honest, I found the linting a bit intense :) Anyway, managed to get them done!

Reply:
good job!! Please let me know if you have a linter that you don't seem to need. If I think so too, can ignore that.

Reply:
I have mixed feeling about the `t.Parallel` enforcement, it's never been a big deal for our tests, but I can see the benefits of always having it.

It might be that I've never been used to a linter in Go so far, I'll raise issue if I feel some should be ignored and we can have the conversation at that time.